Traceback (most recent call last):
  File "/Users/mriki/workspace/prpy/.venv/lib/python3.11/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/Users/mriki/workspace/prpy/.venv/lib/python3.11/site-packages/nbclient/client.py", line 1314, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/mriki/workspace/prpy/.venv/lib/python3.11/site-packages/jupyter_core/utils/__init__.py", line 165, in wrapped
    return loop.run_until_complete(inner)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/mriki/.rye/py/cpython@3.11.8/lib/python3.11/asyncio/base_events.py", line 654, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/Users/mriki/workspace/prpy/.venv/lib/python3.11/site-packages/nbclient/client.py", line 709, in async_execute
    await self.async_execute_cell(
  File "/Users/mriki/workspace/prpy/.venv/lib/python3.11/site-packages/nbclient/client.py", line 1062, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/Users/mriki/workspace/prpy/.venv/lib/python3.11/site-packages/nbclient/client.py", line 918, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
# *** test ***
sample_vector = list(range(100))
y = 7
mod_filter(sample_vector,y)
------------------


[0;31m---------------------------------------------------------------------------[0m
[0;31mNameError[0m                                 Traceback (most recent call last)
Cell [0;32mIn[2], line 4[0m
[1;32m      2[0m sample_vector [38;5;241m=[39m [38;5;28mlist[39m([38;5;28mrange[39m([38;5;241m100[39m))
[1;32m      3[0m y [38;5;241m=[39m [38;5;241m7[39m
[0;32m----> 4[0m [43mmod_filter[49m[43m([49m[43msample_vector[49m[43m,[49m[43my[49m[43m)[49m

Cell [0;32mIn[1], line 3[0m, in [0;36mmod_filter[0;34m(X, y)[0m
[1;32m      1[0m [38;5;28;01mdef[39;00m [38;5;21mmod_filter[39m(X:[38;5;28mlist[39m, y:[38;5;28mint[39m):
[0;32m----> 3[0m     [38;5;28;01mreturn[39;00m [43mresult[49m

[0;31mNameError[0m: name 'result' is not defined

